homeassistant/components/senseme/switch.py

```diff
@@ -0,0 +1,138 @@
+from dataclasses import dataclass
+from typing import Any, cast
+
+from aiosenseme import SensemeFan
+from aiosenseme.device import SensemeDevice
+
+from homeassistant import config_entries
+from homeassistant.components.switch import (
+    SwitchDeviceClass,
+    SwitchEntity,
+    SwitchEntityDescription,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import SensemeEntity
+
+
+@dataclass
+class SenseMESwitchEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[SensemeFan], bool]
+    set_fn: Callable[[SensemeFan, bool], None]
+
+
+@dataclass
+class SenseMESwitchEntityDescription(
+    SwitchEntityDescription, SenseMESwitchEntityDescriptionMixin
+):
+    """Describes SenseME switch entity."""
+
+
+def _set_sleep_mode(device: SensemeDevice, value: bool) -> None:
+    device.sleep_mode = value
+
+
+def _set_motion_fan_auto(device: SensemeDevice, value: bool) -> None:
+    device.motion_fan_auto = value
+
+
+def _set_motion_light_auto(device: SensemeDevice, value: bool) -> None:
+    device.motion_light_auto = value
+
+
+FAN_SWITCHS = [
+    # Turning on sleep mode will disable Whoosh
+    SenseMESwitchEntityDescription(
+        key="sleep_mode",
+        name="Sleep Mode",
+        value_fn=lambda device: cast(bool, device.sleep_mode),
+        set_fn=_set_sleep_mode,
+    ),
+    SenseMESwitchEntityDescription(
+        key="motion_fan_auto",
+        name="Motion",
+        value_fn=lambda device: cast(bool, device.motion_fan_auto),
+        set_fn=_set_motion_fan_auto,
+    ),
+]
+
+FAN_LIGHT_SWITCHES = [
+    SenseMESwitchEntityDescription(
+        key="motion_light_auto",
+        name="Light Motion",
+        value_fn=lambda device: cast(bool, device.motion_light_auto),
+        set_fn=_set_motion_light_auto,
+    ),
+]
+
+LIGHT_SWITCHES = [
+    SenseMESwitchEntityDescription(
+        key="sleep_mode",
+        name="Sleep Mode",
+        value_fn=lambda device: cast(bool, device.sleep_mode),
+        set_fn=_set_sleep_mode,
+    ),
+    SenseMESwitchEntityDescription(
+        key="motion_light_auto",
+        name="Motion",
+        value_fn=lambda device: cast(bool, device.motion_light_auto),
+        set_fn=_set_motion_light_auto,
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: config_entries.ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up SenseME fans."""
+    device = hass.data[DOMAIN][entry.entry_id]
+    descriptions: list[SenseMESwitchEntityDescription] = []
+
+    if device.is_fan:
+        descriptions.extend(FAN_SWITCHS)
+        if device.has_light:
+            descriptions.extend(FAN_LIGHT_SWITCHES)
+    elif device.is_light:
+        descriptions.extend(LIGHT_SWITCHES)
+
+    async_add_entities(
+        [HASensemeSwitch(device, description) for description in descriptions]
```

Comment:
We can drop the brackets and make it a generator expression instead of a list comprehension.

Reply:
https://github.com/home-assistant/core/pull/63774